File: fathersDay/Dsa/Day 20 (48. Rotate Image)/Solution.py
class Solution:
    def rotate(self, matrix: list[list[int]]) -> None:
        matrix[:] = [list(row) for row in zip(*matrix[::-1])]
        return matrix
                
solution=Solution( )
matrix = [[1,2,3],[4,5,6],[7,8,9]]
solution.rotate(matrix)
print(matrix)  # Output: [[7, 4, 1], [8, 5, 2], [9, 6, 3]]

File: fathersDay/Dsa/Day 20 (48. Rotate Image)/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
    void rotate(std::vector<std::vector<int>>& matrix) {
        int n = matrix.size();

        // Transpose the matrix
        for (int i = 0; i < n; ++i) {
            for (int j = i; j < n; ++j) {
                std::swap(matrix[i][j], matrix[j][i]);
            }
        }

        // Reverse each row
        for (int i = 0; i < n; ++i) {
            std::reverse(matrix[i].begin(), matrix[i].end());
        }
    }
};

int main() {
    Solution solution;
    std::vector<std::vector<int>> matrix = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
    solution.rotate(matrix);
    for (const auto& row : matrix) {
        for (int val : row) {
            std::cout << val << " ";
        }
        std::cout << std::endl;
    }
    // Expected Output:
    // 7 4 1 
    // 8 5 2 
    // 9 6 3 
    return 0;
}
